src/Config/Config.ts

```diff
@@ -75,4 +75,63 @@ export default class Config implements IConfigParams {
     this.util = new W3Util(this.web3);
     this.economicStrategy = params.economicStrategy;
   }
+
+  public clientSet (): boolean {
+    return typeof this.client === 'string';
+  }
+
+  public async awaitClientSet (): Promise<any> {
+    return new Promise((resolve) => {
+      if (this.clientSet()) {
+        resolve(true);
+      } else {
+        return this.awaitClientSet();
+      }
+    })
+  }
+  
+  public async getConnectedClient(): Promise<any> {
+    if (!this.web3) {
+      return;
+    }
+    return new Promise((resolve) => {
+      try {
+        const method = 'parity_defaultAccount';
+        this.web3.currentProvider.sendAsync(
+          {
+            jsonrpc: '2.0',
+            method,
+            params: [],
+            id: 0x0a7
+          }, async (err: Error) => {
+            if (!err && !this.clientSet()) {
+              this.client = 'parity';
+              resolve();
+            }
+          }
+        )
+      } catch (e) {
+        return;
+      }
+
+      try {
+        const method = 'net_version';
```

Comment:
We should use a geth specific method here instead since `net_version` is common between parity and geth and perhaps other clients.